packages/text-area/src/vaadin-text-area-mixin.js

```diff
@@ -168,7 +169,11 @@ export const TextAreaMixin = (superClass) =>
 
       const inputHeight = input.scrollHeight;
       if (inputHeight > input.clientHeight) {
-        input.style.height = `${inputHeight}px`;
+        // Subtract padding to get the actual content height
+        const paddingTop = parseInt(inputStyles.paddingTop) || 0;
+        const paddingBottom = parseInt(inputStyles.paddingBottom) || 0;
+        const contentHeight = inputHeight - paddingTop - paddingBottom;
+        input.style.height = `${contentHeight}px`;
```

Comment:
Could setting `box-sizing: border-box` have the same result?

Reply:
Indeed 👍